### PR TITLE
#20 update event handler

### PR DESCRIPTION
**Pull request by nmohan101**
Changes:

1. Renamed Application directory to bin 
2. Added a new directory lib and moved asynctimer.py, getip.py into it. Also added log.py and sql.py to this
3. Updated client.py, server.py, event_handler.py and capture.py import log instead of init in individual files
4. Moved SQL class into its own file (sql.py) from event_handler.py. Updated event_handler so its easier to add new tables if needed in the future. 
5. Updated install script to accommodate new directories. 
6. Update client.py to not error out if PiCamera is not physically installed. (Allow Sim mode)

### CODE REVIEW

File: bin/capture.py [hunk 1]
```diff
@@ -16,6 +16,12 @@
 import logging
```
Comment (nmohan101) [inline bin/capture.py]: This can be removed as its repetitive. from log import log takes care of this.

File: lib/log.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env python
+
+"""log.py: This library faciliates the logging throughput the application.
```
Comment (nmohan101) [inline lib/log.py]: typo.

File: Application/event_handler.py [hunk 3]
```diff
@@ -27,15 +27,15 @@
...
-
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), os.path.pardir)) + "/lib")
+from log import Log
+from sql import SQL
 
 #---------------------------------------------------#
 #                   Constants                       #
 #---------------------------------------------------#
 UID = pwd.getpwuid(os.getuid()).pw_uid
-DBNAME = "/srv/PhotoNetwork/PhotoNetwork.db"
```
Comment (codemonkey111) [inline Application/event_handler.py]: What's up here?  Why did this get removed?  Is it defined in the 'Log' file?

File: Application/event_handler.py [hunk 4]
```diff
@@ -93,112 +93,12 @@ def _create_struct(self):
...
                 tb_index_data += [{"tb_index": tb_index, "tb_name": tb_name}]
-            logger.debug("Requesting Database creation with the following: db_name = {}, tables = {}".format(db_name, tables))
-            sql = SQL(db_path, tables)
+            log.debug("Requesting Database creation with the following: db_name = {}, tables = {}".format(db_name, tables))
+            sql = SQL(db_path, tables, log)
             db_index_data = [{"db_index":db_index, "db_name":db_name, "db_path": db_path, "tables": tb_index_data}]
             all_index += db_index_data
         return db_data, all_index, sql
 
-class SQL(object):
```
Comment (codemonkey111) [inline Application/event_handler.py]: This whole section, what is it? why remove it? What did it do and does it do now?

File: Application/event_handler.py [hunk 5]
```diff
@@ -317,25 +217,9 @@ def read_fifo():
     parser.add_argument('-v', '--verbosity', action = "store_true",  
                         help = "Enter -v for verbosity")
     args = parser.parse_args()
-
```
Comment (codemonkey111) [inline Application/event_handler.py]: This is the section that you moved to it's own file, correct?

File: Application/event_handler.py (hunk 3 above)
Comment (nmohan101) [reply Application/event_handler.py]: Yes. I moved this whole class into a separate file (lib/sql.py)

File: Application/event_handler.py (hunk 4 above)
Comment (nmohan101) [reply Application/event_handler.py]: moved to sql.py. Its the class that is used to read/write to db. Create the db. Basically anything to do with the database is done through this class.

File: Application/event_handler.py (hunk 5 above)
Comment (nmohan101) [reply Application/event_handler.py]: yes.

**Comment by nmohan101** [comment]
Latest commit: 
- Fixed duplication of "import logging" in all files. 
- Fixed typos
- Captures were being overwritten as the file name had only data/time and if 2 captures were taken in the same second file would be over written. (Now each capture has a unique number)
- Updated path for capture.py in srv_capture.c < Eventually this path should come from a config file
- Updated multi-listen function. (rx_data should have been self.rx_data). Removed eval function so now only a string needs to be passed to the multi_listen function.

**Pull request merged**